govcd/tm_content_library_item.go

```diff
@@ -0,0 +1,373 @@
+package govcd
+
+/*
+ * Copyright 2024 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+import (
+	"errors"
+	"fmt"
+	"github.com/vmware/go-vcloud-director/v3/types/v56"
+	"github.com/vmware/go-vcloud-director/v3/util"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+const labelContentLibraryItem = "Content Library Item"
+
+// ContentLibraryItem defines the Content Library Item data structure
+type ContentLibraryItem struct {
+	ContentLibraryItem *types.ContentLibraryItem
+	client             *Client
+}
+
+// wrap is a hidden helper that facilitates the usage of a generic CRUD function
+//
+//lint:ignore U1000 this method is used in generic functions, but annoys staticcheck
+func (g ContentLibraryItem) wrap(inner *types.ContentLibraryItem) *ContentLibraryItem {
+	g.ContentLibraryItem = inner
+	return &g
+}
+
+// CreateContentLibraryItem creates a Content Library Item with the given file located in 'filePath' parameter, which must
+// be an OVA or ISO file.
+func (cl *ContentLibrary) CreateContentLibraryItem(config *types.ContentLibraryItem, filePath string) (*ContentLibraryItem, error) {
+	if _, err := os.Stat(filePath); errors.Is(err, os.ErrNotExist) {
+		return nil, err
+	}
+	cli, err := createContentLibraryItem(cl, config, filePath)
+	if err != nil {
+		if cli == nil || cli.ContentLibraryItem == nil {
+			return nil, err
+		}
+		// We use Name for cleanup because ID may or may not be available
+		return nil, cleanupContentLibraryItemOnUploadError(cl.client, cli.ContentLibraryItem.Name, err)
+	}
+	files, err := getContentLibraryItemPendingFilesToUpload(cli, 1, 10)
```

Comment:
Done. I did not move it outside the method to avoid polluting the module, but it's visible at the top of the function with a (hopefully) meaningful name.

Reply:
Done, moved to constant `retriesForPollingContentLibraryItemFilesToUpload` at the top of this file. It is visible to whole module now, so that's why the "so long, so concise" name (other name suggestions accepted 😅 ).
Also added an explicative comment.